components/backup-stream/src/subscription_track.rs

```diff
@@ -164,10 +164,9 @@ impl SubscriptionTracer {
         let region_id = region.get_id();
         let remove_result = self.0.get_mut(&region_id);
         match remove_result {
+            Some(o) if o.state == SubscriptionState::Removal => false,
             Some(mut o) if if_cond(o.value(), region) => {
```

Comment:
Please do not add condition in match arm, it's very easy to make mistake, IMO. What about 
```rust
match remove_result {
  Some(mut o) => {
    if o.state == SubscriptionState::Removal {
      // ...
    }
    if if_cond(o.value(), region) {
      // ...
    }
  }
} 
```